asp_attendance/models/daily_reporting.py

```diff
@@ -107,20 +127,49 @@ def create_unchecked_daily_reporting_records(self):
         employees = self.env["hr.employee"].search([])
         for employee in employees:
             weekday = today.weekday()
-            working_day = employee.resource_calendar_id.attendance_ids.search([
-                ("dayofweek", "=", weekday)
-            ],limit=1)
+            working_day = employee.resource_calendar_id.attendance_ids.filtered(lambda x: int(x.dayofweek) == weekday)
             existing_attendance = self.env["daily.reporting"].search([
-                        ("employee_id", "=", employee.id),
-                        ("date", "=", today),  
-                ])
+                ("employee_id", "=", employee.id),
+                ("date", "=", today),  
+            ])
             if not existing_attendance:
+                approved_leave = self.env["hr.leave"].search([
+                    ("employee_id", "=", employee.id),
+                    ("state", "=", "validate"),
+                    ("request_date_from", "<=", today),
+                    ("request_date_to", ">=", today),
+                ], limit=1)
                 if not working_day:
                     self.env["daily.reporting"].create({
                         "employee_id": employee.id,
                         "is_not_working_day": True,
                     })
                 else:
-                    self.env["daily.reporting"].create({
-                        "employee_id": employee.id,
-                    })
\ No newline at end of file
+                    if approved_leave:
```

Comment:
Less indentation, you can remove `else` and use `elif`
```suggestion
                    elif approved_leave:
```